ARCHITECTURE.md

```diff
@@ -1,113 +1,280 @@
+
+#### Building a navigation mesh
+
+The project is using `NavMeshComponents`. This means direct building from the Navigation window will not give the desired results. Instead find a `NavMeshComponent` in the given scene e.g. a __NavMeshSurface__ and use the __Bake__ button of that script. Also make sure that there is always only one navmesh file per scene. Navmesh files are stored in a folder with the same name as the corresponding scene. You can recognize them based on their icon in the editor. They follow the naming pattern "NavMesh-<name-of-creating-object.asset>"
+
+## Noteworthy architectural patterns and decisions
+
+### Dependency Injection
+
+We use [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection) pattern, with our library of choice being [VContainer](https://vcontainer.hadashikick.jp/).
+
+DI allows us to clearly define our dependencies in code, as opposed to using static access, pervasive singletons or scriptable object references (aka Scriptable Object Architecture). Code is easy to version-control and comparatively easy to understand for a programmer, as opposed to Unity YAML-based objects, such as scenes, scriptable object instances and prefabs.
+
+DI also allows us to circumvent the problem of cross-scene references to common dependencies, even though we still have to manage the lifecycle of MonoBehaviour-based dependencies by marking them with DontDestroyOnLoad and destroying them manually when appropriate.
+
+> __Note__:
+>
+> `ApplilcationController` inherits from the `VContainer`'s `LifetimeScope` - a class that serves as a dependency injection scope and bootstrapper, where we can bind dependencies. Scene-specific State classes inherit from `LifetimeScope` too.
+> 
+> In Inspector we can choose a parent scope for any `LifetimeScope`s - it’s useful to set a cross-scene reference to some parent scope (most commonly `ApplicationController`). This allows us to bind our scene-specific dependencies, and still have easy access to the global dependencies of the `ApplicationController` in our State-specific version of a `LifetimeScope` object.
+
+### Client/Server code separation
+
+Original problem: reading code with client and server code mixing together adds complexity and makes it easier to make mistakes. That code will often run in a single context, either client or server.
+
+Boss Room explored different client-server code separation approaches. For readers that have been following us since the beginning, we decided in the end to revert our initial client/server/shared assemblies to a more classic domain driven assembly architecture, but still keeping certain more complex classes with a client/server separation.
+
+Our initial thinking was separating assemblies by client and server would allow for easier porting to Dedicated Game Server (DGS) afterward. You’d only need to strip a single assembly to make sure that code only runs when necessary.
```

Comment:
```suggestion
Our initial thinking was that separating assemblies by client and server would allow for easier porting to Dedicated Game Server (DGS) afterward; you’d only need to strip a single assembly to make sure that code only runs when necessary.
```